install/kubernetes/cilium/templates/cilium-ingress-service.yaml

```
@@ -18,10 +18,12 @@ spec:
   - name: http
     port: 80
     protocol: TCP
+    nodePort: {{ .Values.ingressController.service.insecureNodePort }}
```

This value is nil by default - is it correct to add this field if the value is nil? Should we add the field conditionally, e.g. something like:

```suggestion
{{- if .Values.ingressController.service.insecureNodePort }}
    nodePort: {{ .Values.ingressController.service.insecureNodePort }}
{{- end }}
```

Same for the secureNodePort below.

---

Nil should be the same as unset if it’s not already set iirc.

---

This is something that we need to test or otherwise have confidence in (e.g. by checking the reference docs) before merging.

But I did quickly test test this on a local kind cluster for you and it indeed seems that nil is a valid value if the field is required to be absent (e.g. type ClusterIP, or leaving it absent for auto-assignment), so I'm fine with it as is.

---

Yeah, this is just based on how I've done something similar in other helm charts in the past.